apps/condo/domains/resident/schema/RegisterResidentService.test.js

```diff
@@ -2,16 +2,256 @@
-const { makeClientWithNewRegisteredAndLoggedInUser, makeClientWithStaffUser } = require('@condo/domains/user/utils/testSchema')
-const { registerResidentByTestClient, Resident } = require('@condo/domains/resident/utils/testSchema')
+
+const { registerNewOrganization, makeClientWithRegisteredOrganization } = require('@condo/domains/organization/utils/testSchema/Organization')
+
+const { buildingMapJson } = require('@condo/domains/property/constants/property')
 const { Property } = require('@condo/domains/property/utils/testSchema')
 const { buildFakeAddressAndMeta } = require('@condo/domains/property/utils/testSchema/factories')
+const { createTestProperty, makeClientWithResidentAccessAndProperty } = require('@condo/domains/property/utils/testSchema')
+
+const { registerResidentByTestClient, Resident } = require('@condo/domains/resident/utils/testSchema')
+
+const { makeClientWithResidentUser } = require('@condo/domains/user/utils/testSchema')
+const { makeClientWithNewRegisteredAndLoggedInUser, makeClientWithStaffUser } = require('@condo/domains/user/utils/testSchema')
+
+describe('RegisterResidentService connections', () => {
+    it('connects new property with matched address to existing orphan residents (no other props)', async () => {
+        const { address, addressMeta } = buildFakeAddressAndMeta(true)
+
+        const userClient = await makeClientWithResidentUser()
+
+        await registerResidentByTestClient(userClient, { address: addressMeta.value, addressMeta })
+
+        const organizationClient = await makeClientWithRegisteredOrganization()
+
+        // remove flat number from address for organization
+        const orgAddressMeta = { ...addressMeta, value: address }
+        const propertyData = { address, addressMeta: orgAddressMeta, map: buildingMapJson }
+        const [property] = await createTestProperty(organizationClient, organizationClient.organization, propertyData)
+        const [resident] = await Resident.getAll(userClient, { id: userClient.id })
+
+        expect(get(resident, 'organization.id')).toEqual(organizationClient.organization.id)
+        expect(get(resident, 'property.id')).toEqual(property.id)
+    })
+
+    it('connects new property with matched address to existing residents, ignores deleted and younger props', async () => {
+        const { address, addressMeta } = buildFakeAddressAndMeta(true)
+        const userClient = await makeClientWithResidentUser()
+
+        const organizationClient = await makeClientWithRegisteredOrganization()
+        const organizationClient1 = await makeClientWithRegisteredOrganization()
+        const organizationClient2 = await makeClientWithRegisteredOrganization()
+
+        // remove flat number from address for organization
+        const orgAddressMeta = { ...addressMeta, value: address }
+        const propertyPayload = { address, addressMeta: orgAddressMeta, map: buildingMapJson }
```

Comment:
why you can't just use addressMeta here?

Reply:
because `buildFakeAddressAndMeta(true)`, suddenly, puts full address with unit name into `addressMeta.value` and not in `address` and when creating property I don't want unit name to be included in `address` and `addressMeta.value`